### DISCUSSION TITLE
Questions about table formatting

### DISCUSSION THREAD

**Discussion by fadeli1** [eclipse-birt/birt#1251] [Q&A]
I drew a table, the cell is a dynamic text, the contentType is html.

When there are multiple columns, there is no vertical center alignment.

![image](https://user-images.githubusercontent.com/1914030/227911627-bb1eece0-a5bb-4116-961e-538b181a54bd.png)

the contentType is other，center vertically.

sometimes I need to display superscript or subscript, so I have to use html type, like this <SUB>2</SUB>

**Comment by hvbtup** [comment]
?
The items in your image *are* centered vertically.
If you want to display the text centered across two cells, then you have to merge the cells.

**Comment by fadeli1** [comment]
birt rptdesign:
![image](https://user-images.githubusercontent.com/1914030/227967921-5d02b192-ff2e-4129-980b-36b1e87d0ac7.png)

word:
![image](https://user-images.githubusercontent.com/1914030/227968512-ea93cdd3-18d6-4d1c-961b-6574acd8b2b1.png)

hope table2, the content of the red box is also vertically centered, it is strange that it will appear at the bottom of the cell

**Comment by fadeli1** [comment]
I may have found the problem, I open word, use xml mode to open, remove
![image](https://user-images.githubusercontent.com/1914030/227980156-240bd155-e82b-4243-bbcc-ee063e56d184.png)

the contents of table2 are correctly centered vertically
![image](https://user-images.githubusercontent.com/1914030/227980302-9bc4f19a-afde-4aed-b9b9-8a650bf3a7d8.png)

but don't know how to change the source code

**Comment by hvbtup** [comment]
Probably inside your table cell you have two items. The upper one is empty, but it is stil there (did you notice the little line-feed indicator?). You have to hide it if it is empty using the visibility expression.

**Comment by fadeli1** [comment]
here is my structure:
![image](https://user-images.githubusercontent.com/1914030/227988158-78bda7cf-4b17-4a85-91c0-f07ef5c7e7e3.png)

**Comment by fadeli1** [comment]
There should only be one

**Comment by hvbtup** [comment]
Then probably the empty paragraph is inside your html.

**Comment by fadeli1** [comment]
html is this, right?
![image](https://user-images.githubusercontent.com/1914030/227994172-6921c78a-9bf0-4a82-82f2-902ac6755812.png)

**Reply by hvbtup** [reply]
No. This is a dynamic text item containing the text "contentType -> html".

Whether the dynamic text item is HTML or plain text cannot be seen here, that is a property of the item.
It seems like either you misunderstand some concepts of BIRT design, or that I misunderstand your issue.

It would help if you would provide a minimal example report which others can test.

**Reply by fadeli1** [reply]
[new_report.zip](https://github.com/eclipse/birt/files/11080850/new_report.zip)

**Reply by hvbtup** [reply]
This report works as expected. I set a fixed row height of 20mm for both table rows and the report output is vertical centered:

Report Design:
![grafik](https://user-images.githubusercontent.com/7644776/228164189-ab02f99d-be87-461c-9ac9-068c66b2dff8.png)

DOCX output:
![grafik](https://user-images.githubusercontent.com/7644776/228164272-3c25463e-6207-4505-99e7-c4f44e687ada.png)

**Reply by fadeli1** [reply]
The actual situation is that the content: "contentType -> html contentType -> html" has more lines or just one line, it is dynamic

**Reply by fadeli1** [reply]
I tested your solution, there is still a little difference, maybe my birt version is not the latest

doc
![image](https://user-images.githubusercontent.com/1914030/228175265-f7e61d00-90cc-45e9-ac62-ac3d8f62849e.png)

docx
![image](https://user-images.githubusercontent.com/1914030/228175341-4e07d055-cc3a-4532-b232-3380dc61d978.png)

**Reply by fadeli1** [reply]
我准备修改DocEmitterImpl.java文件尝试下，165行
```
context.setIsAfterTable(true);
```
=>
```
// context.setIsAfterTable(true);
```

**Comment by hvbtup** [comment]
I recommend to use the DOCX emitter (ZIP-based file format). The DOC format (proprietary file format) is outdated anyway.
And the DOC emitter cannot be selected in the Designer UI.
Anyway, both file formats work as expected: The content is vertically centered.

**Reply by fadeli1** [reply]
I used another way, modifying the source code

when contentType -> html, do not output the following content
![image](https://user-images.githubusercontent.com/1914030/228497987-bef27b02-4202-4352-88e2-06552e80df2c.png)

I know it's not the right approach, but this worked for me

final effect
![image](https://user-images.githubusercontent.com/1914030/228580983-d95624c3-34c1-44af-9d3f-04e69a2b8d36.png)

not sure if there will be problems later
